tests/integration/managed_shared_ptr_tests.cpp

```diff
@@ -0,0 +1,360 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC and CHAI
+// project contributors. See the CHAI LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////
+#include <type_traits>
+#include "camp/defines.hpp"
+#include "chai/ChaiMacros.hpp"
+#include "chai/ExecutionSpaces.hpp"
+#include "chai/ManagedSharedPtr.hpp"
+#include "chai/SharedPtrManager.hpp"
+#include "gtest/gtest.h"
+#include "umpire/ResourceManager.hpp"
+
+#define GPU_TEST(X, Y)              \
+  static void gpu_test_##X##Y();    \
+  TEST(X, Y) { gpu_test_##X##Y(); } \
+  static void gpu_test_##X##Y()
+
+#include "chai/config.hpp"
+#include "chai/ArrayManager.hpp"
+#include "chai/ManagedArray.hpp"
+#include "chai/managed_ptr.hpp"
+#include "chai/ManagedSharedPtr.hpp"
+
+#include "../src/util/forall.hpp"
+
+// Standard library headers
+#include <cstdlib>
+
+#if defined(CHAI_ENABLE_CUDA)
+inline void gpuErrorCheck(cudaError_t code, const char *file, int line, bool abort=true)
```

Comment:
Aren't these already defined somewhere? If not, we should add a test utils header for these.